src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindAndUpdateIntegrationTest.java

```diff
@@ -415,6 +402,91 @@ public void findByIdAndUnset() {
           .body("data.docs[0]", jsonEquals(expected));
     }
 
+    @Test
+    @Order(2)
+    public void findByIdAndUnsetNested() {
+      insertDoc(
+          """
+                  {
+                    "_id": "update_doc_unset_nested",
+                    "array": [
+                        137,
+                        { "x" : 1, "y" : 2 }
+                    ],
+                    "subdoc" : {
+                        "x" : 5,
+                        "y" : -19
+                    }
+                  }
+              """);
+
+      // NOTE: we mix actual working removals and ones that won't; it is not an error
+      // to try to $unset properties that do not (or sometimes cannot) exist. They just
+      // have no effect.
+      //
+      // Ones that do have effect are:
+      //
+      // * array.0   -> remove first entry, replace with null
+      // * array.1.x -> remove property 'x' from second array element (object)
+      // * subdoc.y  -> remove subdoc property 'y'
+      String json =
+          """
+                   {
+                      "findOneAndUpdate": {
+                        "filter" : {"_id" : "update_doc_unset_nested"},
+                        "update" : {"$unset" : {
```

Comment:
Would also be nice to test that `$unset: { _id: 1 }` throws an error

Reply:
I think that is already tested in another test (unit test I think). But I can add one here just for completeness' sake.